File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/user/dto/UserDeleteResponse.java
package com.portfolio.auctionmarket.domain.user.dto;

import com.portfolio.auctionmarket.domain.user.entity.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserDeleteResponse {
    private String email;

    public static UserDeleteResponse from(User entity) {
        return UserDeleteResponse.builder()
                .email(entity.getEmail())
                .build();
    }
}

File: AuctionMarket/auction-app-user/src/main/kotlin/org/example/auctioncommon/domain/user/dto/UserDeleteResponse.kt
package org.example.auctioncommon.domain.user.dto

import org.example.auctioncommon.domain.user.entity.User

data class UserDeleteResponse(
    val email: String
)

fun User.toDeleteDto() : UserDeleteResponse {
    return UserDeleteResponse(
        email = this.email
    )
}